pkg/registry/controller/strategy.go

```diff
@@ -41,6 +42,12 @@ type rcStrategy struct {
 // Strategy is the default logic that applies when creating and updating Replication Controller objects.
 var Strategy = rcStrategy{api.Scheme, api.SimpleNameGenerator}
 
+// DefaultGarbageCollectionPolicy returns Orphan because that was the default
+// behavior before the server-side garbage collection was implemented.
+func (rcStrategy) DefaultGarbageCollectionPolicy() rest.GarbageCollectionPolicy {
+	return rest.OrphanDependents
```

Comment:
Yes. DeleteOptions have the highest priority. The existing finalizers on the object has the second highest priority. This default policy has the lowest priority.